### PR TITLE
Improve AngularJS performance with one-time binding for static strings

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
Improves the run speed of all Angular pages in CiviCRM by making use of the one-time binding feature for static strings being passed through ts().

Before
----------------------------------------
Angular pages slower

After
----------------------------------------
Angular pages faster

Technical Details
----------------------------------------
One-time binding prevents unnecessary $watch expressions on values that will never change, making the digest loop faster.
See https://docs.angularjs.org/guide/expression#one-time-binding

Comments
---------
I also included a test case for strings in one-time binding expressions.

### CODE REVIEW

File: ang/crmCaseType/list.html [hunk 1]
```diff
@@ -13,9 +13,9 @@ <h1 crm-page-title>{{ts('Case Types')}}</h1>
   <table class="display">
     <thead>
     <tr>
-      <th>{{ts('Title')}}</th>
-      <th>{{ts('Name')}}</th>
-      <th>{{ts('Description')}}</th>
+      <th>{{:: ts('Title') }}</th>
+      <th>{{:: ts('Name') }}</th>
+      <th>{{:: ts('Description') }}</th>
       <th>{{ts('Enabled?')}}</th>
```
Comment (demeritcowboy) [inline ang/crmCaseType/list.html:19]: I'm guessing a regex thing for line 19?

File: ang/crmCxn/PermTable.html [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <table>
   <thead>
   <tr>
-    <th>{{ts('Entity')}}</th>
+    <th>{{:: ts('Entity') }}</th>
     <th>{{ts('Action(s)')}}</th>
     <th>{{ts('Filter(s)')}}</th>
```
Comment (demeritcowboy) [inline ang/crmCxn/PermTable.html:6]: Another regex gotcha?

File: ang/crmCaseType/list.html (hunk 1 above)
Comment (colemanw) [reply ang/crmCaseType/list.html]: Expanded the regex to include any character inside ts() as long as there are no variable substitutions.

File: ang/crmCxn/PermTable.html (hunk 2 above)
Comment (colemanw) [reply ang/crmCxn/PermTable.html]: Fixed with new regex.

**Comment by colemanw** [comment]
@demeritcowboy my regex was a bit too conservative. I've updated it to cast a wider net, should still be excluding any expression that uses variable substitution. Thanks for reviewing :)

**Comment by demeritcowboy** [comment]
* General standards
    * [`r-explain`] __PASS__
    * [`r-user`] __PASS__
    * [`r-doc`] __PASS__
    * [`r-run`] __PASS__
      * I have no data on performance increase but everything seems to work.
      * I also checked what happens if you have visited the page and then change the language whether this causes it to get "stuck" on the old binding but it uses the new lang ok.
        * There's a pre-existing issue if you use the language switcher while on an angular page but I think that's been logged elsewhere already, just can't find it. If I can't find it will create ticket.
* Developer standards
    * [`r-tech`] __PASS__
    * [`r-code`] __PASS__
      * As an aside, I was also curious about the use of `ts('HTML')`. I think it's technically correct and best to leave it, just noting that the internet suggests that other languages use HTML, SQL, etc as-is and a quick look at the civi repo suggest the same, even for languages like Russian with a different alphabet. Just noting, since this PR is about performance.
    * [`r-maint`] __PASS__
    * [`r-test`] __PASS__

**Pull request merged**